File: DTeacher/app/src/main/java/com/dreamwalker/knu2018/dteacher/Adapter/WriteDrugAdapter.java
package com.dreamwalker.knu2018.dteacher.Adapter;

import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.TextView;

import com.dreamwalker.knu2018.dteacher.Model.Person;
import com.dreamwalker.knu2018.dteacher.R;

import java.util.List;

/**
 * Created by KNU2017 on 2018-02-09.
 */

public class WriteDrugAdapter extends RecyclerView.Adapter<WriteDrugAdapter.PersonViewHolder> {
    List<Person> people;
    OnItemClickListener mItemClickListener;

    public WriteDrugAdapter(List<Person> persons) {
        this.people = persons;
    }

    @Override
    public PersonViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_drug_card, viewGroup, false);
        return new PersonViewHolder(v);
    }

    @Override
    public void onBindViewHolder(PersonViewHolder personViewHolder, int i) {
        personViewHolder.personName.setText(people.get(i).name);
        personViewHolder.personAge.setText(people.get(i).age);
        personViewHolder.personPhoto.setImageResource(people.get(i).photoId);
        personViewHolder.cbSelected.setChecked(people.get(i).isSelected());

    }

    public void setItemSelected(int position, boolean isSelected) {
        if (position != -1) {
            people.get(position).setSelected(isSelected);
            notifyDataSetChanged();
        }
    }

    public List<Person> getPeople() {
        return people;
    }

    @Override
    public int getItemCount() {
        return people.size();
    }

    @Override
    public void onAttachedToRecyclerView(RecyclerView recyclerView) {
        super.onAttachedToRecyclerView(recyclerView);
    }

    public void setOnItemClickListener(final OnItemClickListener mItemClickListener) {
        this.mItemClickListener = mItemClickListener;
    }

    public void setPeople(List<Person> people) {
        this.people = people;
    }


    public interface OnItemClickListener {
        void onItemClick(View view, int position);
    }

    public class PersonViewHolder extends RecyclerView.ViewHolder implements
            View.OnClickListener {

        CardView cv;
        TextView personName;
        TextView personAge;
        ImageView personPhoto;
        CheckBox cbSelected;

        PersonViewHolder(View itemView) {
            super(itemView);
            itemView.setOnClickListener(this);
            cv = (CardView) itemView.findViewById(R.id.cv);
            personName = (TextView) itemView.findViewById(R.id.person_name);
            personAge = (TextView) itemView.findViewById(R.id.person_age);
            personPhoto = (ImageView) itemView.findViewById(R.id.person_photo);
            cbSelected = (CheckBox) itemView.findViewById(R.id.cbSelected);
        }

        @Override
        public void onClick(View v) {
            if (mItemClickListener != null) {
                mItemClickListener.onItemClick(v, getPosition());
            }
        }
    }


}

File: DTeacher_kt/app/src/main/java/com/dreamwalker/knu2018/dteacher/Adapter/WriteDrugAdapter.kt
package com.dreamwalker.knu2018.dteacher.Adapter

import android.support.v7.widget.CardView
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.TextView

import com.dreamwalker.knu2018.dteacher.Model.Person
import com.dreamwalker.knu2018.dteacher.R

/**
 * Created by KNU2017 on 2018-02-09.
 */

class WriteDrugAdapter(var people: List<Person>) : RecyclerView.Adapter<WriteDrugAdapter.PersonViewHolder>() {
    internal var mItemClickListener: OnItemClickListener? = null

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): PersonViewHolder {
        val v = LayoutInflater.from(viewGroup.context).inflate(R.layout.item_drug_card, viewGroup, false)
        return PersonViewHolder(v)
    }

    override fun onBindViewHolder(personViewHolder: PersonViewHolder, i: Int) {
        personViewHolder.personName.text = people[i].name
        personViewHolder.personAge.text = people[i].age
        personViewHolder.personPhoto.setImageResource(people[i].photoId)
        personViewHolder.cbSelected.isChecked = people[i].isSelected

    }

    fun setItemSelected(position: Int, isSelected: Boolean) {
        if (position != -1) {
            people[position].isSelected = isSelected
            notifyDataSetChanged()
        }
    }

    override fun getItemCount(): Int {
        return people.size
    }

    override fun onAttachedToRecyclerView(recyclerView: RecyclerView?) {
        super.onAttachedToRecyclerView(recyclerView)
    }

    fun setOnItemClickListener(mItemClickListener: OnItemClickListener) {
        this.mItemClickListener = mItemClickListener
    }


    interface OnItemClickListener {
        fun onItemClick(view: View, position: Int)
    }

    inner class PersonViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

        internal var cv: CardView
        internal var personName: TextView
        internal var personAge: TextView
        internal var personPhoto: ImageView
        internal var cbSelected: CheckBox

        init {
            itemView.setOnClickListener(this)
            cv = itemView.findViewById<View>(R.id.cv) as CardView
            personName = itemView.findViewById<View>(R.id.person_name) as TextView
            personAge = itemView.findViewById<View>(R.id.person_age) as TextView
            personPhoto = itemView.findViewById<View>(R.id.person_photo) as ImageView
            cbSelected = itemView.findViewById<View>(R.id.cbSelected) as CheckBox
        }

        override fun onClick(v: View) {
            if (mItemClickListener != null) {
                mItemClickListener!!.onItemClick(v, position)
            }
        }
    }


}
